File: blind-75/stacks/valid-parentheses/python/valid_parentheses.py
def valid_parentheses(s: str) -> bool:
  if len(s) % 2 != 0:
    return False

  pair = {")": "(", "}": "{", "]": "["}
  stack = []

  for i in range(len(s)):
    c = s[i]

    if c == "(" or c == "{" or c == "[":
      stack.append(c)
    elif len(stack) == 0:
      return False
    else:
      open = stack.pop()
      if open != pair.get(c): return False

  return len(stack) == 0

File: blind-75/stacks/valid-parentheses/go/valid_parentheses.go
package validparentheses

func ValidParentheses(s string) bool {
	if len(s) % 2 != 0 {
		return false
	}

	pair := map[string]string{
		")": "(",
		"}": "{",
		"]": "[",
	}

	stack := []string{}

	for i := 0; i < len(s); i++ {
		c := s[i]

		if c == '(' || c == '{' || c == '[' {
			stack = append(stack, string(c))
		} else if len(stack) == 0 {
			return false
		} else {
			open := stack[len(stack) - 1]
			stack = stack[:len(stack) - 1]
			
			if open != pair[string(c)] {
				return false
			}
		}
	}

	return len(stack) == 0
}